### PR TITLE
Add actionlint

### LINKED ISSUE #867: Evaluate adding actionlint
[actionlint](https://github.com/rhysd/actionlint) is a static checker for GitHub Actions workflow files. It is already configured in [Contracts for Solidity](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3953).

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes #867.

I think it makes sense to include actionlint as it already found a few things to improve (most notably [SC2086](https://www.shellcheck.net/wiki/SC2086)).

This PR also proposes to remove .codecov and .gitmodules as they are no longer used

### CODE REVIEW

File: .github/workflows/actionlint.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Add problem matchers
+        run: |
+          # https://github.com/rhysd/actionlint/blob/3a2f2c7/docs/usage.md#problem-matchers
+          curl -LO https://raw.githubusercontent.com/rhysd/actionlint/main/.github/actionlint-matcher.json
```
Comment (ericnordelo) [inline .github/workflows/actionlint.yml:16]: Why not having this locally under `.github/actionlint-matcher.json`?

File: .github/workflows/actionlint.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          echo "::add-matcher::actionlint-matcher.json"
+      - uses: docker://rhysd/actionlint:latest
```
Comment (ericnordelo) [inline .github/workflows/actionlint.yml:18]: Does this run the checker automatically?

**Review by ericnordelo — APPROVED** [review]

**Review by ericnordelo — COMMENTED** [review]
Looking good! Left a couple of questions.

File: .github/workflows/actionlint.yml (hunk 2 above)
Comment (andrew-fleming) [reply .github/workflows/actionlint.yml:18]: Yes, indeed

File: .github/workflows/actionlint.yml (hunk 1 above)
Comment (ericnordelo) [reply .github/workflows/actionlint.yml:16]: Security-wise is better to have it locally, right?

**Review by immrsd — APPROVED** [review]
LGTM, good job!

**Pull request merged**